plugin/evm/limitorders/memory_database.go

```diff
@@ -300,6 +301,11 @@ func (db *InMemoryDatabase) UpdateFilledBaseAssetQuantity(quantity *big.Int, ord
 	defer db.mu.Unlock()
 
 	limitOrder := db.OrderMap[orderId]
+	if limitOrder == nil {
+		log.Error("In UpdateFilledBaseAssetQuantity - orderId does not exist in the database", "orderId", orderId.Hex())
+		metrics.GetOrRegisterCounter("update_filled_base_asset_quantity_order_id_not_found", nil).Inc(1)
```

Comment:
i suggest declaring/registering this metric in plugin/evm/orderbook/metrics.go file. If the code never reaches this point, then this metric will not be registered and won't be returned in metrics API. If we register this in metrics.go file then it will always be returned in the API and then we can have a chart/alert based on it on grafana.